saleor/graphql/order/dataloaders.py

```diff
@@ -63,6 +63,9 @@ class OrderLinesByOrderIdLoader(DataLoader):
     def batch_load(self, keys):
         lines = (
             OrderLine.objects.using(self.database_connection_name)
+            # Products are prefetched, as they are needed in all
+            # financial calculations related to order lines
+            .select_related("variant__product")
```

Comment:
Not sure about this, this will load the products and variants always, even if you're just loading plain order lines, which adds additional queries. Can't we instead use the product/variants dataloader where we are sure that we need products/variants? We can easily chain results of multiple dataloaders which we do in many resolvers.

Reply:
Can't you load products with a data loader to a separate queryset and just pass it down to where it's needed?

Reply:
I'll just revert back to prefetching in `fetch_order_prices_if_expired` - this function will be called only once for every single unconfirmed order change, so while the benchmark might look like a yellow flag, in reality this is going to be cached pretty much permanently.